### ISSUE TITLE
A @Bean of Enum injection

### ISSUE THREAD

**Issue by kliushnichenko** [avaje/avaje-inject#710]
Given the following config:

```java
public enum State {
    ON, OFF
}

@Factory
public class Configuration {

    @Bean
    public State state() {
        return State.ON;
    }

}
```

Later on, inject the `State` bean somewhere. The compilation is successful, but when the app is starting up:

` Injecting null for ... when creating class .... - potential beans to inject: []. Check for missing module? [ missing beanScopeBuilder.modules() ]. If it is expected to be externally provided, missing beanScopeBuilder.bean() ?`

did I miss something, or it's not supported?

**Comment by SentryMan** [comment]
Just tried it and it works for me.

**Comment by SentryMan** [comment]
perhaps you are running into #702? Perhaps try 10.5-RC3

**Comment by kliushnichenko** [comment]
Well, the issue arises only when there is run-time `String` bean registration in addition to the Factory example above, like here https://github.com/jooby-project/jooby/blob/74bab2f968362c73e8d1e3af8f0e53aec28e985d/modules/jooby-avaje-inject/src/main/java/io/jooby/avaje/inject/AvajeInjectModule.java#L94.

reproducible with 10.5-RC3

**Comment by SentryMan** [comment]
I see the issue will have a fix soon. In the mean time you can add a `@Named` qualifier to make it work.
```java
@Factory
public class Configuration {

    @Bean
    @Named("state")
    public State state() {
        return State.ON;
    }

}

**Comment by SentryMan** [comment]
Fixed in `10.5-RC4`
